orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/loadbalancer/UpsertLoadBalancersTask.groovy

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2016 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.clouddriver.tasks.loadbalancer
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.netflix.spinnaker.orca.DefaultTaskResult
+import com.netflix.spinnaker.orca.ExecutionStatus
+import com.netflix.spinnaker.orca.RetryableTask
+import com.netflix.spinnaker.orca.TaskResult
+import com.netflix.spinnaker.orca.clouddriver.KatoService
+import com.netflix.spinnaker.orca.clouddriver.tasks.AbstractCloudProviderAwareTask
+import com.netflix.spinnaker.orca.pipeline.model.Stage
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+
+/*
+ * Task that can create multiple load balancers
+ */
+@Component
+class UpsertLoadBalancersTask extends AbstractCloudProviderAwareTask implements RetryableTask {
+
+  static final String CLOUD_OPERATION_TYPE = "upsertLoadBalancer"
+
+  @Override
+  long getBackoffPeriod() {
+    return 2000
+  }
+
+  @Override
+  long getTimeout() {
+    return 300000
+  }
+
+  @Autowired
+  KatoService kato
+
+  @Autowired
+  ObjectMapper mapper
+
+  @Override
+  TaskResult execute(Stage stage) {
+    String cloudProvider = getCloudProvider(stage)
+    String account = getCredentials(stage)
```

Comment:
Okay, now I'm looking at the `CreateDeployTask`, which seems to be what you based this task on, and it's doing the same thing you're doing. So one or more of the following are true:
 * a deploy stage doesn't really work across multiple providers (account seems fine, as the deploy task does not assume the same account)
 * the `cloudProvider` argument in `kato.requestOperations` doesn't do what I would think it would (i.e. fail if it's mismatched with the actual cloud provider the operation should operate against)
 * everything - including this PR - works fine because the underlying code it's calling is a little confusing.

I do think the `account` needs to be extracted per operation, since it could very easily change between configured load balancers.